pkg/proxystorage/proxy.go

```diff
@@ -733,6 +733,84 @@ func (p *ProxyStorage) NodeReplacer(ctx context.Context, s *parser.EvalStmt, nod
 			return n, nil
 		}
 
+	// BinaryExprs *can* be sent untouched to downstreams assuming there is no actual interaction between LHS/RHS
+	// these are relatively rare -- as things like `sum(foo) > 2` would *not* be viable as `sum(foo)` could
+	// potentially require multiple servergroups to generate the correct response.
+	// For now we'll start a VERY basic check for a `VectorSelector` and either of (`NumberLiteral` or `StringLiteral`)
+	case *parser.BinaryExpr:
```

Comment:
From looking through the code and thinking about it -- the only scenarios I can think of where this is "safe" are if it is a literal and one of the following:
- VectorSelector
- AggregateExpr: Max, Min


So I'd like to add those cases here; add some tests; and clean this up prior to merging :)